hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go

```diff
@@ -191,7 +195,7 @@ func (a Azure) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _ *hy
 			},
 		)
 	}
-
```

Comment:
any reason to not do this in the v2Component definition via InjectAvailabilityProberContainer? that would also impact all other providers

Reply:
Only that that wasn't how it was done for AWS capi provider.  I can refactor to get it  into the v2 component.